R2Northstar/NorthstarLauncher
# mods(Safe I/O): Only allow creating files with whitelisted filetypes

Only allow creating files with whitelisted filetypes to prevent writing `.bat` or `.vb` files with Safe I/O that could then be called using some exploit (e.g. #674), we should prevent writing file types that by default are interpreted as executable.

Whitelisting was chosen over blacklisting as we can always extend the list and don't have to worry about potentially forgetting to blacklist a certain filetype.

Of course this doesn't prevent writing a bash script to a `.txt` file and then somehow getting Windows to interpret it as a batch file.

Reading files is not restricted to filetypes as we are primarily concerned with creating files here.

Supersedes #675

~~Completely untested atm.~~

### Testing instructions:

1. Use the following mod: [GeckoEidechse.Safe-IO-Testing.zip](https://github.com/user-attachments/files/17837293/GeckoEidechse.Safe-IO-Testing.zip)
2. Go into private match and start a match
3. Enable cheats (`sv_cheats 1`)
4. Open console and run
    ```
    script GeckoSaveFile()
    ```
5. Go to `<profile folder>\save_data\GeckoEidechse.Safe-IO-Testing\`
6. Check that only `test1.txt` exists but no `test2.bat` or `test3`.
